[BACKEND] Transformed TritonGPUToLLVMPass to TritonGPUToLLVM in amd backend

zahimoud opened the pull request:
Changes:

- Extracted SPMD, Print, ControlFlow patterns to separate files
- Deleted TritonGPUToLLVM
- Renamed TritonGPUToLLVMPass to TritonGPUToLLVM
- Extracted ControlFlow, and GetProgramId, FuncOp patterns to common lib

zahimoud commented:
Running amd backend tests [link](https://github.com/openai/triton/actions/runs/8209868584/job/22456224074)

ThomasRaoux commented on include/triton/Conversion/TritonGPUToLLVM/TargetInfoBase.h:
```diff
@@ -22,6 +22,8 @@ class TargetInfoBase {
                            Value val, int i) const = 0;
   virtual Value shuffleIdx(Location loc, ConversionPatternRewriter &rewriter,
                            Value val, Value i) const = 0;
+  virtual Value llGetPid(Location loc, ConversionPatternRewriter &rewriter,
```
I would rename it to `programId()`

ThomasRaoux commented on include/triton/Conversion/TritonGPUToLLVM/PatternTritonGPUOpToLLVM.h:
```diff
@@ -47,6 +47,11 @@ void populateViewOpToLLVMPatterns(LLVMTypeConverter &typeConverter,
                                   RewritePatternSet &patterns,
                                   PatternBenefit benefit);
 
+void populatePrintOpToLLVMPattern(LLVMTypeConverter &typeConverter,
```
Can we enable the tests for print?

ThomasRaoux reviewed (approved):
Looks good!

zahimoud commented:
- Renamed `targetInfo.llGetPid()` to `targetInfo.programId()`
- Moved back PrintOp pattern to nvidia's backend, deleted conversion from amd backend as it's currently not supported
